spec/controllers/bikes_controller_spec.rb

```diff
@@ -1501,30 +1509,9 @@ def expect_bike_transferred_but_unclaimed(bike, user)
         expect(bike.bike_organization_ids).to match_array([organization.id, organization_2.id])
         expect(bike.editable_organizations.pluck(:id)).to eq([organization_2.id])
       end
-      context "empty organization_ids_can_edit_claimed" do
```

Comment:
I think this is an important test case? This tests that you can remove all the editable organizations

Reply:
IIRC this was an instance of [this issue](https://github.com/rails/rails/issues/27215#issuecomment-263972010) but I'll do a manual test to see if we still have the expected behavior.

> This is the expected behavior. There is no way to encode an empty array using an HTML form (e.g. Content-Type: url-form-encoded). The reason your tests passed in Rails 4.2 is because controller tests did not encode their parameters, they simply passed the hash through directly. In Rails 5 it encodes them. If your controller cares about empty arrays, it's likely that you are dealing with JSON requests. You can do that in your test with as: :json. If you're just dealing with form input, you will never receive an empty array.

Reply:
Well, we need to solve this test case.

First solution off the top of my head (provided the issue you remarked upon is correct): add a hidden field to that form which makes the controller know the organization ids are being passed - so if no ids are passed, it defaults to an empty array.

Let's create an issue for this and resolve after this is merged, I would prefer to ship this first.

Reply:
But - can you put this test back in, and mark it pending, before merging?

Reply:
Agreed on validating the use case is still satisfied — no change there, it still works, afaict:

![demo](https://user-images.githubusercontent.com/4433943/73384461-51525080-4299-11ea-8486-958b1e86e022.gif)

We have a `:organization_ids_can_edit_claimed_present` hidden field on the form presently:

```haml
-# app/views/bikes/edit_groups.html.haml L13-15 (8ae68fdc)

- if @bike.bike_organizations.any?
  = hidden_field_tag :organization_ids_can_edit_claimed_present, true
  .related-fields.no-divider-row
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/8ae68fdc/app/views/bikes/edit_groups.html.haml#L13-L15)]</sup>

The removed test case can't pass in Rails 5, and on `master` it's a spuriously passing test. I think this test is validating the use case in question?:

```rb
# spec/controllers/bikes_controller_spec.rb L1512-1526 (992e9c53)

context "organization_ids_can_edit_claimed_present" do
  it "updates the bike with the allowed_attributes, marks no organizations can edit claimed" do
    put :update, params: { id: bike.id, bike: allowed_attributes, organization_ids_can_edit_claimed_present: "1" }
    expect(response).to redirect_to edit_bike_url(bike)
    expect(assigns(:bike)).to be_decorated
    bike.reload
    expect(bike.hidden).to be_falsey
    allowed_attributes.except(*skipped_attrs).each do |key, value|
      pp value, key unless bike.send(key) == value
      expect(bike.send(key)).to eq value
    end
    expect(bike.bike_organization_ids).to match_array([organization.id, organization_2.id])
    expect(bike.editable_organizations.pluck(:id)).to eq([])
  end
end
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/992e9c53/spec/controllers/bikes_controller_spec.rb#L1512-L1526)]</sup>

Reply:
I'll add it back in just in case though 👍

Reply:
Ha. Historical Seth was as smart as current Seth, too bad I don't remember what he did.

Thinking about it - I bet this test was a test I left in because I'm lazy - I probably initially tested it this way, and then figured out that the test didn't represent how things worked, added the new test but didn't delete the old, unrepresentative test. 

Basically, I'm a forgetful idiot. Feel free to delete. Closing the issue.